lib/monitor.js

```diff
@@ -214,3 +218,33 @@ internals.forOwn = (obj, func) => {
         func(key, obj[key]);
     }
 };
+
+internals.validateStream = (throwError, lastIndex, index, stream, reporterName, streamName) => {
+
+    if (stream instanceof Stream.Transform) {
```

Comment:
if the inheritance doesn't break the prototype chain there shouldn't be an issue.
If it does than it's not following any standard prototypale inheritance (So not using `Object.create` or `class extends`).
The code ignores a stream if it can't be sure it's wrong (like Duplex streams or anything that doesn't extend a native node stream). 
I can throw a warning in those situations as well but it might introduce some false positives